torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -320,6 +320,13 @@ class FullyShardedDataParallel(nn.Module):
             the near future. It allows users to enable two different backward_prefetch
             algorithms to help backward communication and computation overlapping.
             Pros and cons of each algorithm is explained in the class ``BackwardPrefetch``.
+        mixed_precision: (Optional[MixedPrecision]): A ``MixedPrecision`` instance
+            describing the mixed precision training config to be used. ``MixedPrecision``
+            supports configuring paramter, buffer, and gradient communication dtype. Note
+            that only floating point data is cast to the reduced precision. This allows
+            users potential memory saving and training speedup while trading off
```

Comment:
Do we have reference on how to understand the tradeoff here, in particular the reduces precision?

Reply:
It will vary based on workload, hw type, and mixed precision config (how parameters, buffers, and gradient communication can be configured differently). Currently other than suggesting bfloat16 as a first step if the HW supports it otherwise fp16 + ShardedGradScaler, it will rely on user experimentation and testing which could possibly be improved by autotuning framework in the future.